applications/FluidDynamicsApplication/custom_processes/distance_modification_process.h

```diff
@@ -295,12 +304,22 @@ class DistanceModificationProcess : public Process
 
     void RecoverOriginalDistance()
     {
-        for(unsigned int i=0; i<mModifiedDistancesIDs.size(); i++)
+        #pragma omp parallel
         {
-            unsigned int nodeId = mModifiedDistancesIDs[i];
-            mrModelPart.GetNode(nodeId).FastGetSolutionStepValue(DISTANCE) = mModifiedDistancesValues[i];
+            const int ThreadId = OpenMPUtils::ThisThread();
+            const std::vector<unsigned int> LocalModifiedDistancesIDs = mModifiedDistancesIDs[ThreadId];
+            const std::vector<double> LocalModifiedDistancesValues = mModifiedDistancesValues[ThreadId];
+
+            for(unsigned int i=0; i<LocalModifiedDistancesIDs.size(); ++i)
+            {
+                const unsigned int nodeId = LocalModifiedDistancesIDs[i];
+                mrModelPart.GetNode(nodeId).FastGetSolutionStepValue(DISTANCE) = LocalModifiedDistancesValues[i];
```

Comment:
you don't really have another option here, still i would like to make clear from a "didactic" point of view that the GetNode function has a cost that is "logN" where N is the number of nodes. 
this means that the cost of this loop is NlogN (not N as one might expect)